src/common/support_utils.cpp

```diff
@@ -15,65 +15,65 @@ char *eofstr(char *str) {
     return (str + strlen(str));
 }
 
-void block2hex(char *str, uint8_t *pdata, size_t length) {
+void block2hex(char *str, uint32_t str_size, uint8_t *pdata, size_t length) {
     for (; length > 0; length--)
-        sprintf(eofstr(str), "%02X", *(pdata++));
+        snprintf(eofstr(str), str_size - strlen(str), "%02X", *(pdata++));
```

Comment:
this construct is suspicious - why str_size - strlen(str)?
it should be at least explained in a comment

Reply:
resolved temporarily, will be refactored (to avoid computing strlen all over again)